File: php/features.php
<?php
declare(strict_types=1);

// Insult SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class InsultFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new InsultBaseFeature();
            case "test":
                return new InsultTestFeature();
            default:
                return new InsultBaseFeature();
        }
    }
}

File: py/insult_sdk/features.py
# Insult SDK feature factory

from insult_sdk.feature.base_feature import InsultBaseFeature
from insult_sdk.feature.test_feature import InsultTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InsultBaseFeature(),
        "test": lambda: InsultTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
